src/stream/src/executor/barrier_align.rs

```diff
@@ -33,10 +35,34 @@ pub trait AlignedMessageStream = futures::Stream<Item = AlignedMessageStreamItem
 #[derive(Debug, EnumAsInner, PartialEq)]
 pub enum AlignedMessage {
     Barrier(Barrier),
+    Watermark(Watermark),
     Left(StreamChunk),
     Right(StreamChunk),
 }
 
+fn get_watermarks_to_send(
+    left_cached_watermarks: &mut VecDeque<Watermark>,
+    right_cached_watermarks: &mut VecDeque<Watermark>,
+) -> StreamExecutorResult<Vec<Watermark>> {
+    let mut ret = vec![];
+    while let (Some(left_front), Some(right_front)) = (
+        left_cached_watermarks.front(),
+        right_cached_watermarks.front(),
+    ) {
+        match left_front.partial_cmp(right_front).ok_or(anyhow::anyhow!(
```

Comment:
May directly use `cmp` here, as it will panic on incompatible data types.

Reply:
> May directly use `cmp` here, as it will panic on incompatible data types.

fixed